profile/get.go

```diff
@@ -72,7 +72,7 @@ func GetProfile(source profileSource, id, linkedPath string, reset bool) ( //nol
 
 		// Get from database.
 		if !reset {
-			profile, err = findProfile(linkedPath)
+			profile, err = findProfile(id, linkedPath)
 			// Check if the profile is special and needs a reset.
 			if err == nil && specialProfileNeedsReset(profile) {
 				profile = getSpecialProfile(id, linkedPath)
```

Comment:
L90-100:

There seems to be a bug already in here, as line 89 should call `New()` with the given source.

This is also the point where we would need to add checks on which profiles can be created automatically in which case if we follow along with my suggestion below. An additional flag to the function of `idByUser` could specify whether the profile ID was supplied by the user and may only be automatically created in the `user` scope.
Also, we can probably ditch the checks on line 111 for now, until we start using layered profiles.